code/datums/components/egg_layer.dm

```diff
@@ -0,0 +1,84 @@
+/**
+ * # egg layer component!
+ *
+ * Component that manages how many eggs to lay, what can be fed to the mob to make them lay more, and what is actually laid.
+ * Since the only real interaction with the component is an attackby, the nice part is that we're able to make this an atom level proc.
+ * egg_layer will loudly fail if you do not provide it the arguments, as to encourage explicicy(?)
+ */
+/datum/component/egg_layer
+	/// item laid by the mob
+	var/egg_type
+	/// items that can be fed to the mob to make it lay more eggs
+	var/list/food_types
+	/// messages sent when fed
+	var/list/feedMessages
+	/// messages sent when laying an egg
+	var/list/layMessages
+	/// kind of a weird boolean, but basically egg layers usually need to make their eggs process to let them grow. and some don't want that to happen too!
+	var/process_eggs
+	/// how many eggs left to lay
+	var/eggs_left = 0
+	/// used to check for possibly too much egg laying going on (aka past a certain amount of chickens, stop making more chickens from eggs)
+	var/datum/callback/egg_animal_callback
```

Comment:
This is a weird name and its documentation doesn't do it much justice.

Reply:
true and fair